### PR TITLE
Use Buffer.UnsafeCursor to avoid big byte[] copy.

### PR DESCRIPTION
**Pull request by NightlyNexus**
We can put the bytes right into the ByteBuffer, if I understand correctly.

### CODE REVIEW

**Comment by NightlyNexus** [comment]
@JakeWharton I'm opening this to confirm my understanding.

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.java [hunk 1]
```diff
@@ -116,8 +116,22 @@ static Bitmap decodeStream(Source source, Request request) throws IOException {
   @SuppressLint("Override")
   private static Bitmap decodeStreamP(Request request, BufferedSource bufferedSource)
       throws IOException {
-    ImageDecoder.Source imageSource =
-        ImageDecoder.createSource(ByteBuffer.wrap(bufferedSource.readByteArray()));
+    bufferedSource.request(Long.MAX_VALUE);
+    Buffer buffer = bufferedSource.getBuffer();
+    long remaining = buffer.size();
+    ByteBuffer byteBuffer = ByteBuffer.allocate((int) remaining);
```
[the remark this answers is not in the document]
Comment (NightlyNexus) [reply picasso/src/main/java/com/squareup/picasso3/BitmapUtils.java]: Oh, true. Wrap doesn't make a copy, I don't think.

[the remark this answers is not in the document]
Comment (JakeWharton) [reply picasso/src/main/java/com/squareup/picasso3/BitmapUtils.java]: Right. It uses the array you bring.

[the remark this answers is not in the document]
Comment (NightlyNexus) [reply picasso/src/main/java/com/squareup/picasso3/BitmapUtils.java]: yep! 👍

**Pull request closed without merging**